File: iasa47471/iasa_jogo/src/jogo/ecr/ComportComp.java
package jogo.ecr;

import java.util.ArrayList;
import java.util.List;

import jogo.personagem.Accao;
import jogo.personagem.Percepcao;

public class ComportComp implements Comportamento {
    private List<Comportamento> comportamentos = new ArrayList<>();

    @Override
    public Accao activar(Percepcao percepcao) {
        return null;
    }
    
}

File: iasa47471/iasa_agente/src/lib/ecr/ComportComp.py
from abc import ABC, abstractmethod
import IComportamento

# Esta classe representa um comportamento composto.
# Um comportamento pode ser composto por outros comportamentos,
# criado assim um comportamento composto.
# Esta classe implementa a interface IComportamento
class ComportComp(IComportamento):
    # Construtor da classe ComportComp
    # Recebe uma lista de comportamentos
    def __init__(self, comportamentos):
        self._comportamentos = comportamentos

    # Ativa uma accao a partir da percepcao recebida
    # Retorna a accao ativada
    def activar(percepcao):
        # TODO
        return None

    # Método abstrato
    # Seleciona uma accao da lista de accoes entregue
    @abstractmethod
    def seleccionar_accao(accoes):
        '''
        Selecionar accao
        @param percepcao: lista de accoes
        @return: acção selecionada
        '''